audioswitch/src/main/java/com/twilio/audioswitch/bluetooth/BluetoothScoJob.kt

```diff
@@ -0,0 +1,57 @@
+package com.twilio.audioswitch.bluetooth
+
+import android.os.Handler
+import androidx.annotation.VisibleForTesting
+import androidx.annotation.VisibleForTesting.PRIVATE
+import com.twilio.audioswitch.android.LogWrapper
+import com.twilio.audioswitch.android.SystemClockWrapper
+import com.twilio.audioswitch.bluetooth.BluetoothDeviceConnectionListener.ConnectionError
+import java.util.concurrent.TimeoutException
+
+internal const val TIMEOUT = 5000L
+private const val TAG = "BluetoothScoManager"
+
+internal abstract class BluetoothScoJob(
+    private val logger: LogWrapper,
+    private val bluetoothScoHandler: Handler,
+    private val systemClockWrapper: SystemClockWrapper
+) {
+
+    @VisibleForTesting(otherwise = PRIVATE)
+    var bluetoothScoRunnable: BluetoothScoRunnable = BluetoothScoRunnable()
+    @VisibleForTesting(otherwise = PRIVATE)
+    var deviceListener: BluetoothDeviceConnectionListener? = null
+
+    protected abstract val scoAction: () -> Unit
+    protected abstract val timeoutError: ConnectionError
+
+    fun executeBluetoothScoJob() {
+        bluetoothScoRunnable = BluetoothScoRunnable()
```

Comment:
Should this line be removed since the field is set at objection construction time?

Reply:
Ah yeah so this line is in place to reset the elapsed time for the next job that runs. One alternative is reseting the elapsed time on line 29 instead with `bluetoothScoRunnable.elapsedTime = 0L`.